File: app/src/main/java/com/example/testapp/test/TestApi.java
package com.example.testapp.test;

import com.bankcomm.commlibrary.http.Call;
import com.bankcomm.commlibrary.http.core.Get;
import com.bankcomm.commlibrary.http.core.Params;


/**
 * https://github.com/anmnight
 * author：anxiao on 2018/8/13 16:56
 * anmnight@qq.com
 */
public interface TestApi {

    @Get("service/getIpInfo.php")
    Call<IpResult> getData(@Params("ip") String ip);
}

File: commlibrary/src/test/java/com/anmnight/commlibrary/TestApi.kt
package com.anmnight.commlibrary

import com.example.testapp.http.Call
import com.example.testapp.http.core.Body
import com.example.testapp.http.core.Get
import com.example.testapp.http.core.Query

/**
 * https://github.com/anmnight
 * author：anxiao on 2018/8/17 15:44
 * anmnight@qq.com
 */
interface TestApi {

    @Get("/user")
    fun user(@Query("id") id: Int, @Body version: Version): Call<TestEntity>

    @Get("/version/all")
    fun version(): Call<Version>
}